@bernhardkaindl I see this message in the job trace of the job you linked above:

```
==> [2021-09-22-08:19:07.970297] numactl: Executing phase: 'autoreconf'
==> [2021-09-22-08:19:12.767781] Error: ProcessError: Command exited with status 1:
    '/usr/bin/bash' './autogen.sh'
See build log for details:
  /tmp/root/spack-stage/spack-stage-numactl-2.0.14-2irmcuao37qogct5eqi65lobebo7a4rs/spack-build-out.txt
Traceback (most recent call last):
  File "/builds/spack/spack/lib/spack/spack/build_environment.py", line 1022, in _setup_pkg_and_run
    return_value = function(pkg, kwargs)
  File "/builds/spack/spack/lib/spack/spack/installer.py", line 1883, in build_process
    return installer.run()
  File "/builds/spack/spack/lib/spack/spack/installer.py", line 1749, in run
    self._real_install()
  File "/builds/spack/spack/lib/spack/spack/installer.py", line 1847, in _real_install
    phase(pkg.spec, pkg.prefix)
  File "/builds/spack/spack/lib/spack/spack/package.py", line 116, in phase_wrapper
    phase(spec, prefix)
  File "/builds/spack/spack/var/spack/repos/builtin/packages/numactl/package.py", line 38, in autoreconf
  File "/builds/spack/spack/lib/spack/spack/util/executable.py", line 225, in __call__
    raise ProcessError('Command exited with status %d:' %
spack.util.executable.ProcessError: Command exited with status 1:
    '/usr/bin/bash' './autogen.sh'
```

Somehow even though that failed, `spack install` exited non-zero so the CI code continued on afterwards, attempting to create a binary package for software that failed to even configure.  Hence, the error message was actually a long way from the bottom where you likely expected to see it.  Sorry about that 😞 

If you browse the artifacts of the failed job, you can find `jobs_scratch_dir/logs/spack-build-out.txt` which in this case has the following contents:

```
==> [2021-09-22-08:19:07.970297] numactl: Executing phase: 'autoreconf'
==> [2021-09-22-08:19:07.971902] '/usr/bin/bash' './autogen.sh'
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'build-aux'.
libtoolize: linking file 'build-aux/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'm4'.
libtoolize: linking file 'm4/libtool.m4'
libtoolize: linking file 'm4/ltoptions.m4'
libtoolize: linking file 'm4/ltsugar.m4'
libtoolize: linking file 'm4/ltversion.m4'
libtoolize: linking file 'm4/lt~obsolete.m4'
Useless use of a variable in void context at /home/software/spack/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spack_path_placeholder__/__spa/linux-ubuntu20.04-ppc64le/gcc-9.3.0/automake-1.16.3-rzfsznblqwmptsoua2hoy6rxgqevce5j/bin/automake line 8405.
autoreconf: automake failed with exit status: 255
```

It makes me think there could be a real problem, and that re-running the pipeline is likely fail in the same way.

🤷 Hope it helps.

--- a/lib/spack/spack/build_systems/autotools.py
+++ b/lib/spack/spack/build_systems/autotools.py
@@ -252,6 +252,18 @@ class AutotoolsPackage(PackageBase):
         if self.force_autoreconf:
             force_remove(self.configure_abs_path)
 
+    def _autoreconf_warning(self, spec, missing):
+        msg = ("Cannot generate configure: missing dependencies {0}.\n\nPlease add "
+               "the following lines to the package:\n\n".format(", ".join(missing)))
+
+        for dep in missing:
+            msg += ("    depends_on('{0}', type='build', when='@{1}')\n"
+                    .format(dep, spec.version))
+
+        msg += "\nUpdate the version (when='@{0}') as needed.".format(spec.version)
+
+        return msg
+
     def autoreconf(self, spec, prefix):
         """Not needed usually, configure should be already there"""
         # If configure exists nothing needs to be done
@@ -261,9 +273,10 @@ class AutotoolsPackage(PackageBase):
         needed_dependencies = ['autoconf', 'automake', 'libtool']
         build_deps = [d.name for d in spec.dependencies(deptype='build')]
         missing = [x for x in needed_dependencies if x not in build_deps]
+
         if missing:
-            msg = 'Cannot generate configure: missing dependencies {0}'
-            raise RuntimeError(msg.format(missing))
+            raise RuntimeError(self._autoreconf_warning(spec, missing))
+
         tty.msg('Configure script not found: trying to generate it')
         tty.warn('*********************************************************')
         tty.warn('* If the default procedure fails, consider implementing *')

--- a/lib/spack/spack/build_systems/autotools.py
+++ b/lib/spack/spack/build_systems/autotools.py
@@ -258,8 +258,9 @@ class AutotoolsPackage(PackageBase):
         if os.path.exists(self.configure_abs_path):
             return
         # Else try to regenerate it
-        autotools = ['m4', 'autoconf', 'automake', 'libtool']
-        missing = [x for x in autotools if x not in spec]
+        needed_dependencies = ['autoconf', 'automake', 'libtool']
+        build_deps = [d.name for d in spec.dependencies(deptype='build')]
+        missing = [x for x in needed_dependencies if x not in build_deps]
         if missing:
             msg = 'Cannot generate configure: missing dependencies {0}'
             raise RuntimeError(msg.format(missing))

